src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,171 @@
+            }
+
+            int stringLength = input.length();
+            int index = Integer.parseInt(input.substring(stringLength - 1)) - 1;
+
+            Task task = taskHandler.retrieveTask(index);
+            taskHandler.delete(index);
+
+            int size = taskHandler.getSize();
+
+            return ui.deleteTask(task, size);
+
+        } else if (input.startsWith("find")) {
+            try {
+                validateInput(input, 5);
+            } catch (UserInputException e) {
+                return ui.printIncompleteCommand(e.getMessage());
+            }
+
+            String[] keywords = input.split(" ");
+            List<Task> filteredList = taskHandler.filterByWord(keywords[1]);
+
+            return ui.printStorageList(filteredList);
+
+        } else {
+            if (!(input.startsWith("todo") || input.startsWith("event") 
+                    || input.startsWith("deadline"))) {
+                return ui.invalidInput();
+            } else if (input.startsWith("todo")) {
+                try {
+                    validateInput(input, 5);
+                } catch (UserInputException e) {
+                    return ui.printIncompleteCommand(e.getMessage());
+
+                }
+
+                String[] moreStrings = input.split(" ", 2);
+
+                Task todo = new Todo(moreStrings[1]);
+                taskHandler.add(todo);
+                int index = taskHandler.getSize();
+
+                return ui.addTask(todo, index);
+            } else if (input.startsWith("deadline")) {
+                try {
+                    validateInput(input, 9);
+                } catch (UserInputException e) {
+                    return ui.printIncompleteCommand(e.getMessage());
+
+                }
+
+                String[] moreStrings = input.split("/", 2);
+                String[] getDescription = moreStrings[0].split(" ", 2);
+
+                String returnBy = moreStrings[1].substring(3);
+
+                Task deadline = new Deadline(getDescription[1], returnBy);
+                taskHandler.add(deadline);
+                int index = taskHandler.getSize();
+                
+                return ui.addTask(deadline, index);
+            } else if (input.startsWith("event")) {
+                try {
+                    validateInput(input, 6);
+                } catch (UserInputException e) {
+                    return ui.printIncompleteCommand(e.getMessage());
+
+                }
+
+                String[] moreStrings = input.split("/");
+                String[] pullStrings = moreStrings[0].split(" ", 2);
+
+                String from = moreStrings[1].substring(4);
+                String to = moreStrings[2].substring(2);
+
+                Task event = new Event(pullStrings[1], from, to);
+                taskHandler.add(event);
+                int index = taskHandler.getSize();
+
+                return ui.addTask(event, index);
+            }
```

Comment:
Nonetheless, I still like how the method is still readable despite its long code